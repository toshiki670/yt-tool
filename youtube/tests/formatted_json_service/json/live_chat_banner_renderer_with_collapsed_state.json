{
  "replayChatItemAction": {
    "actions": [
      {
        "addBannerToLiveChatCommand": {
          "bannerRenderer": {
            "liveChatBannerRenderer": {
              "header": {
                "liveChatBannerHeaderRenderer": {
                  "icon": { "iconType": "KEEP" },
                  "text": {
                    "runs": [
                      { "text": "Pinned by " },
                      {
                        "text": "Test Channel Name"
                      }
                    ]
                  },
                  "contextMenuButton": {
                    "buttonRenderer": {
                      "icon": { "iconType": "MORE_VERT" },
                      "accessibility": { "label": "Chat actions" },
                      "accessibilityData": {
                        "accessibilityData": { "label": "Chat actions" }
                      },
                      "command": {
                        "commandMetadata": {
                          "webCommandMetadata": { "ignoreNavigation": true }
                        },
                        "liveChatItemContextMenuEndpoint": {
                          "params": "Q2g0S0hBb2FRMUJxVjJ0eGRuWjBXVEJFUmxwMlQzZG5VV1JUVjBrd1ZWRWFLU29uQ2hoVlF6WnZURGhuWlV0TlpYbG5iUzAwT1dsTWNuRmxabmNTQzBkUk4xQjJUbEpWVkZwUklBRW9CRElhQ2hoVlEyazRlWGt5ZHpSQ0xWODFSMjE1TlRWTk1VTXhNVkU0QVVnQVVBRSUzRA=="
                        }
                      }
                    }
                  }
                }
              },
              "contents": {
                "liveChatTextMessageRenderer": {
                  "message": {
                    "runs": [
                      {
                        "text": "https://example.com/test/status/123456789...",
                        "navigationEndpoint": {
                          "commandMetadata": {
                            "webCommandMetadata": {
                              "url": "https://www.youtube.com/redirect?event=live_chat&redir_token=MASKED_TOKEN&q=https%3A%2F%2Fexample.com%2Ftest%2Fstatus%2F123456789",
                              "webPageType": "WEB_PAGE_TYPE_UNKNOWN",
                              "rootVe": 83769
                            }
                          },
                          "urlEndpoint": {
                            "url": "https://www.youtube.com/redirect?event=live_chat&redir_token=MASKED_TOKEN&q=https%3A%2F%2Fexample.com%2Ftest%2Fstatus%2F123456789",
                            "target": "TARGET_NEW_WINDOW",
                            "nofollow": true
                          }
                        }
                      }
                    ]
                  },
                  "authorName": { "simpleText": "Test User" },
                  "authorPhoto": {
                    "thumbnails": [
                      {
                        "url": "https://example.com/avatar32.jpg",
                        "width": 32,
                        "height": 32
                      },
                      {
                        "url": "https://example.com/avatar64.jpg",
                        "width": 64,
                        "height": 64
                      }
                    ]
                  },
                  "id": "ChwKGkNQaldrcXZ2dFkwREZadk93Z1FkVFdJMFVR",
                  "timestampUsec": "1747875417862973",
                  "authorBadges": [
                    {
                      "liveChatAuthorBadgeRenderer": {
                        "icon": { "iconType": "MODERATOR" },
                        "tooltip": "Moderator",
                        "accessibility": {
                          "accessibilityData": { "label": "Moderator" }
                        }
                      }
                    }
                  ],
                  "authorExternalChannelId": "UCTestChannelId123456789"
                }
              },
              "actionId": "ChwKGkNNYWU3TGp2dFkwREZhVUlyUVlkWExBbUJR",
              "viewerIsCreator": false,
              "targetId": "live-chat-banner",
              "isStackable": false,
              "backgroundType": "LIVE_CHAT_BANNER_BACKGROUND_TYPE_STATIC",
              "bannerProperties": {
                "autoCollapseDelay": { "seconds": "7" },
                "bannerCollapsedStateEntityKey": "EihDaHdLR2tOTllXVTNUR3AyZEZrd1JFWmhWVWx5VVZsa1dFeEJiVUpSIIsBKAE%3D"
              },
              "bannerType": "LIVE_CHAT_BANNER_TYPE_PINNED_MESSAGE"
            }
          }
        }
      }
    ]
  },
  "videoOffsetTimeMsec": "3327238",
  "isLive": true
}
