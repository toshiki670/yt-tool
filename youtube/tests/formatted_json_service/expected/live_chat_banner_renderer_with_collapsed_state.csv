id,authorExternalChannelId,postedAt,category,authorName,content
ChwKGkNNYWU3TGp2dFkwREZhVUlyUVlkWExBbUJR,,2025-05-22T09:56:57.862973+09:00,Banner,Pinned by Test Channel Name,TextMessageID: 'ChwKGkNQaldrcXZ2dFkwREZadk93Z1FkVFdJMFVR'
ChwKGkNQaldrcXZ2dFkwREZadk93Z1FkVFdJMFVR,UCTestChannelId123456789,2025-05-22T09:56:57.862973+09:00,Text message,Test User,message: 'https://example.com/test/status/123456789'; Moderator
